Fix: Vertical alignment div has no effect for pdf output

MBe-iUS opened the pull request:
## Description
Closes #9109.

This PR fixes the vertical alignment problem in Div's with Latex format. 

The problem was diagnosed in #9109 and related to the change in layout handling.

The fix simply converts the new layout class properties `quarto-layout-valign-{top,center, bottom}` to the Latex pendants `{top,center, bottom}`.

## Checklist

I have (if applicable):

- [x] filed a [contributor agreement](https://github.com/quarto-dev/quarto-cli/blob/main/CONTRIBUTING.md).
- [x] referenced the GitHub issue this PR closes
- [x] updated the appropriate changelog in the PR
- [x] ensured the present test suite passes
- [x] added new tests
- [ ] created a separate documentation PR in [Quarto's website repo](https://github.com/quarto-dev/quarto-web/) and linked it to this PR

MBe-iUS committed:
Fix: Vertical alignment div has no effect for pdf output

MBe-iUS committed:
Inform changelog

MBe-iUS committed:
Handle valign_class is nil

MBe-iUS committed:
Escape lua pattern hyphen

MBe-iUS committed:
Adapt test file snapshot

Default setting in function VAlignValidated for minipage is "top". This is now preset and needs to be reflected in the snapshot file.

kernie committed:
New test for valign=bottom

MBe-iUS committed:
Merge branch 'quarto-dev:main' into main

MBe-iUS committed:
Merge branch 'main' into main

MBe-iUS committed:
Merge branch 'main' into main

MBe-iUS committed:
Merge branch 'main' into main

MBe-iUS committed:
Merge branch 'main' into main

cderv committed:
Merge branch 'main' into pr-13658

cderv committed:
Move changelog in correct place

cderv reviewed (approved):
Looks good yes ! 

I'll handle the rest before merging, as we are closed to 1.9 release

cderv committed:
Typo in comment

cderv committed:
kLayoutVAlign already removed

in panellayout.lua

cderv committed:
Fix layout-valign tests to match positional argument in minipage

Convert snapshot tests (8354, 9109) to ensureLatexFileRegexMatches
and update 7262 regex to expect [b] position. Add center alignment
test coverage.

cderv committed:
Add test for layout-valign with plain markdown content

Covers the exact scenario from #9109: div syntax with
layout-valign on plain text, no code execution needed.
